mpcs/src/basefold/encoding/rs.rs

```diff
@@ -374,519 +280,98 @@ where
-    }
-
-    #[test]
-    fn test_naive_fft_with_rate() {
-        use rand::rngs::OsRng;
-        let num_vars = 5;
-        let rate_bits = 1;
-
-        let poly: Vec<GoldilocksExt2> = (0..(1 << num_vars))
-            .map(|_| GoldilocksExt2::random(&mut OsRng))
-            .collect();
-        let mut poly2 = vec![GoldilocksExt2::ZERO; poly.len() * (1 << rate_bits)];
-        poly2.as_mut_slice()[..poly.len()].copy_from_slice(poly.as_slice());
-        let mut poly2 = FieldType::Ext(poly2.clone());
-
-        let naive = naive_fft::<GoldilocksExt2>(&poly, 1 << rate_bits, Goldilocks::GENERATOR);
-
-        let root_table = fft_root_table(num_vars + rate_bits);
-        coset_fft::<GoldilocksExt2>(&mut poly2, Goldilocks::GENERATOR, rate_bits, &root_table);
-
-        let poly2 = match poly2 {
-            FieldType::Ext(coeffs) => coeffs,
-            _ => panic!("Wrong field type"),
-        };
-        assert_eq!(naive, poly2);
-    }
-
-    #[test]
-    fn test_ifft() {
-        let num_vars = 5;
-
-        let poly: Vec<GoldilocksExt2> =
-            (0..(1 << num_vars)).map(GoldilocksExt2::from_u64).collect();
-        let mut poly = FieldType::Ext(poly);
-        let original = poly.clone();
-
-        let root_table = fft_root_table(num_vars);
-        fft::<GoldilocksExt2>(&mut poly, 0, &root_table);
-        ifft::<GoldilocksExt2>(&mut poly, 0, &root_table);
-
-        assert_eq!(original, poly);
-    }
-
-    #[test]
-    fn prover_verifier_consistency() {
-        type Code = RSCode<RSCodeDefaultSpec>;
-        let pp: RSCodeParameters<GoldilocksExt2> = Code::setup(10);
-        let (pp, vp) = Code::trim(pp, 10).unwrap();
-        for level in 0..(10 + <Code as EncodingScheme<GoldilocksExt2>>::get_rate_log()) {
-            for index in 0..(1 << level) {
-                let (naive_x0, naive_x1, naive_w) =
-                    Code::folding_coeffs_naive(level, index, pp.full_message_size_log);
-                let (p_x0, p_x1, p_w) = Code::prover_folding_coeffs(&pp, level, index);
-                let (v_x0, v_x1, v_w) = Code::verifier_folding_coeffs(&vp, level, index);
-                // assert_eq!(v_w * (v_x1 - v_x0), GoldilocksExt2::ONE);
-                // assert_eq!(p_w * (p_x1 - p_x0), GoldilocksExt2::ONE);
-                assert_eq!(
-                    (v_x0, v_x1, v_w, p_x0, p_x1, p_w),
-                    (naive_x0, naive_x1, naive_w, naive_x0, naive_x1, naive_w),
-                    "failed for level = {}, index = {}",
-                    level,
-                    index
-                );
-            }
-        }
-    }
-
-    #[test]
-    fn test_rs_codeword_folding() {
-        test_codeword_folding::<GoldilocksExt2, RSCode<RSCodeDefaultSpec>>();
-    }
+    use super::*;
 
     type E = GoldilocksExt2;
     type F = Goldilocks;
     type Code = RSCode<RSCodeDefaultSpec>;
+    use crate::BasefoldRSParams;
 
     #[test]
-    pub fn test_colinearity() {
+    pub fn test_message_codeword_linearity() {
```

Comment:
This new unittest has tested the essential functionality: 
`basefold.encode(raw_message.fold(1-r, r)) ?= codeword.fold(1-r, r)`

All the other tests are not that matters, e.g. fold on reverse bit codeword etc, as we dont need this functionality